kaldb/src/main/java/com/slack/kaldb/server/KaldbIndexer.java

```diff
@@ -50,8 +50,9 @@
 
   private final IndexingChunkManager<LogMessage> chunkManager;
 
-  public static LogMessageTransformer getLogMessageTransformer() {
-    String dataTransformerConfig = KaldbConfig.get().getIndexerConfig().getDataTransformer();
+  public static LogMessageTransformer getLogMessageTransformer(
+      KaldbConfigs.IndexerConfig indexerConfig) {
+    String dataTransformerConfig = indexerConfig.getDataTransformer();
```

Comment:
Why are we passing the `indexerConfig` if we only use the `dataTransformer` string? This should only require the string config to be passed.

Reply:
Passing in most general type for now since there may be more params in future,

Reply:
As above, let's pass in the most specific argument. This case especially since this code will be moving out of the indexer entirely.

Reply:
Done.